### PR TITLE
Added form UI support for editable artifact upload size

### PR DESCRIPTION
**Pull request by JanBucko**
# Pull Request Description

This pull request solves part of Strongbox/Strongbox#1819

# Acceptance Test

* [x] Building the code with `ng build` still works.
* [ ] Building the code with `ng test` still works.
* [x] Building the code with `ng e2e` still works.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [x] Yes, please see Strongbox/Strongbox#1819
  * [ ] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

# Code Review And Pre-Merge Checklist

* [x] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [x] I have performed a self-review of my own code.
* [ ] I have commented my code in hard-to-understand areas.
* [x] My changes generate no new warnings.
* [ ] I have added tests that prove my fix is effective or that my feature works.
* [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html [hunk 1]
```diff
@@ -51,6 +51,16 @@
...
+
+                  <div fxLayout.lt-md="column" fxLayout="row" fxLayoutGap="10%">
+                    <mat-form-field>
+                      <input matInput placeholder="Max upload size" formControlName="maxUploadSize">
+                      <mat-hint>Maximum artifact size that will be accepted by the server.</mat-hint>
+                      <mat-error>
+                        <form-field-errors formControlName="maxFileSize"></form-field-errors>
+                      </mat-error>
+                    </mat-form-field>
+                  </div>
```
Comment (carlspring) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html]: Please, follow the same formatting as the rest of our code. Thanks! :)

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts [hunk 2]
```diff
@@ -55,6 +55,20 @@ export class ManageSettingsComponent implements OnInit, OnDestroy {
...
+                  invalidSize: 'Max file upload size should be: "unlimited" or for example 100MB'
+                };
+
+                if (control.invalid) {
+                  return message;
+                }
+
+                return null;
+              }
+            ]),
```
Comment (carlspring) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts]: As above.

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html (hunk 1 above)
Comment (JanBucko) [reply src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html]: Was that about the missing tab ? I am sorry I havent noticed

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts (hunk 2 above)
Comment (JanBucko) [reply src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts]: I hope i fixed it :S

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html [hunk 3]
```diff
@@ -51,6 +51,16 @@
...
+
+                    <div fxLayout.lt-md="column" fxLayout="row" fxLayoutGap="10%">
+                      <mat-form-field>
+                        <input matInput placeholder="Max upload size" formControlName="maxUploadSize">
+                        <mat-hint>Maximum artifact size that will be accepted by the server.</mat-hint>
+                        <mat-error>
+                          <form-field-errors formControlName="maxFileSize"></form-field-errors>
+                        </mat-error>
+                      </mat-form-field>
+                    </div>
```
Comment (carlspring) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html]: Cosmetic: Would you mind following the same formatting as the rest of the code (around your changes) here and below? Thanks! :)

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts [hunk 4]
```diff
@@ -55,6 +55,21 @@ export class ManageSettingsComponent implements OnInit, OnDestroy {
...
```
Comment (steve-todorov) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts:72]: Cosmetic: Follow the code style which is `4 white spaces` (you seem to be using 2)

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html [hunk 5]
```diff
@@ -51,6 +51,16 @@
...
                     </div>
+
+                    <div fxLayout.lt-md="column" fxLayout="row" fxLayoutGap="10%">
+                        <mat-form-field>
+                            <input matInput placeholder="Max upload size" formControlName="maxUploadSize">
+                            <mat-hint>Maximum artifact size that will be accepted by the server.</mat-hint>
+                            <mat-error>
+                                <form-field-errors formControlName="maxFileSize"></form-field-errors>
+                            </mat-error>
+                      </mat-form-field>
```
Comment (steve-todorov) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.html:62]: Cosmetic: Missing 4 white spaces.
```suggestion
                          </mat-form-field>
```

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts [hunk 6]
```diff
@@ -55,6 +55,21 @@ export class ManageSettingsComponent implements OnInit, OnDestroy {
                     return null;
                 }
             ]),
+            maxUploadSize: new FormControl('unlimited', [
```
Comment (steve-todorov) [inline src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts:58]: The key must match what the backend expects as well as the `formControlName` you set in the `manage-settings.component.html`. Currently there's a mismatch, which causes the build failure. 

image

You also need to add the same property in the `ServerSettings` under `server-settings.module.ts` for better IDE autocomplete support.

**Review by steve-todorov — CHANGES_REQUESTED** [review]
I've `rebased` your PR and you will need to clone it again and do an `npm install` before continuing since the current `master` now uses Angular 10. 

There are some minor fixes you need to apply and we can merge. :)

File: src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts (hunk 6 above)
Comment (steve-todorov) [reply src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts:58]: One more thing - please follow the indentation in the file. We're using 4 spaces across the entire project. :)

**Comment by steve-todorov** [comment]
Did you close this by accident ? :)

**Pull request closed without merging**